code/datums/traits/good.dm

```diff
@@ -0,0 +1,83 @@
+//predominantly positive traits
+//this file is named weirdly so that positive traits are listed above negative ones
+
+/datum/trait/alcohol_tolerance
+	name = "Alcohol Tolerance"
+	desc = "You become drunk more slowly and suffer fewer drawbacks from alcohol."
+	value = 1
+	mob_trait = TRAIT_ALCOHOL_TOLERANCE
+	gain_text = "<span class='notice'>You feel like you could drink a whole keg!</span>"
+	lose_text = "<span class='danger'>You don't feel as resistant to alcohol anymore. Somehow.</span>"
+
+
+
+/datum/trait/freerunning
+	name = "Freerunning"
+	desc = "You're great at quick moves! You can climb tables more quickly."
+	value = 2
+	mob_trait = TRAIT_FREERUNNING
+	gain_text = "<span class='notice'>You feel lithe on your feet!</span>"
+	lose_text = "<span class='danger'>You feel clumsy again.</span>"
+
+
+
+/datum/trait/light_step
+	name = "Light Step"
+	desc = "You walk with a gentle step, making stepping on sharp objects quieter and less painful."
+	value = 1
+	mob_trait = TRAIT_LIGHT_STEP
+	gain_text = "<span class='notice'>You walk with a little more lithenessk.</span>"
+	lose_text = "<span class='danger'>You start tromping around like a barbarian.</span>"
+
+
+
+/datum/trait/night_vision
+	name = "Night Vision"
+	desc = "You can see slightly more clearly in full darkness than most people."
+	value = 1
+	mob_trait = TRAIT_NIGHT_VISION
+	gain_text = "<span class='notice'>The shadows seem a little less dark.</span>"
+	lose_text = "<span class='danger'>Everything seems a little darker.</span>"
+
+/datum/trait/night_vision/on_spawn()
+	var/mob/living/carbon/human/H = trait_holder
```

Comment:
So is it livings or humans, typecheck

Reply:
Traits have a human requirement by default in their base type that ensures that on_spawn won't be called if it's not a human, since the trait won't be assigned.

Reply:
But I see some of these traits could be assigned perfectly well to livings, rather than just humans.

Reply:
They're coded around human requirements due to how the code handles them.

Now that I consider it, the system could likely stand to use significant improvement...